mso/resource_mso_schema_site_anp_epg_domain.go

```diff
@@ -308,32 +336,69 @@ func resourceMSOSchemaSiteAnpEpgDomainCreate(d *schema.ResourceData, m interface
 	siteId := d.Get("site_id").(string)
 	anpName := d.Get("anp_name").(string)
 	epgName := d.Get("epg_name").(string)
-	domainType := d.Get("domain_type").(string)
-	domainName := d.Get("dn").(string)
 	deployImmediacy := d.Get("deploy_immediacy").(string)
 	resolutionImmediacy := d.Get("resolution_immediacy").(string)
 
-	var DN, microSegVlanType, portEncapVlanType, vlanEncapMode, switchingMode, switchType, enhancedLagpolicyName, enhancedLagpolicyDn string
+	var DN, microSegVlanType, portEncapVlanType, vlanEncapMode, switchingMode, switchType, enhancedLagpolicyName, enhancedLagpolicyDn, domainType, domainName string
 	var microSegVlan, portEncapVlan float64
 	var allowMicroSegmentation bool
 
-	if domainType == "vmmDomain" {
-		DN = fmt.Sprintf("uni/vmmp-VMware/dom-%s", domainName)
+	domainNameDnOld, ok_oldName := d.GetOk("dn")
+	tempVarName, ok_name := d.GetOk("domain_name")
+	tempVarDn, ok_dn := d.GetOk("domain_dn")
+
+	if !ok_oldName && !ok_name && !ok_dn {
+		return fmt.Errorf("domain_dn or domain_name in association with domain_type and vmm_domain_type if applicable are required.")
+	} else if ok_name {
+		domainName = tempVarName.(string)
+	} else if ok_oldName {
+		domainName = domainNameDnOld.(string)
+	} else if ok_dn {
+		DN = tempVarDn.(string)
+		re_domain := regexp.MustCompile("uni/(.*)-(.*)")
```

Comment:
re_domain := regexp.MustCompile("uni/(.*?)-(.*)")

Reply:
The above function didn't produce desired result.